### DISCUSSION TITLE
Fix Fauna integration demo

### DISCUSSION THREAD

**Discussion by LL782** [vercel/community#809] [Idea]
The Vercel integration for Fauna offers a demo via this page:
https://vercel.com/integrations/fauna#template

Here's a link to the demo:
https://fauna-nextjs-guestbook-eight.vercel.app/

I've tried this before and after the weekend and I get the same error each time. I assume it will be easy for anyone to reproduce but let me know if I can help with more details.

Below is a screenshot:
<img width="912" alt="Screenshot 2022-08-29 at 22 18 47" src="https://user-images.githubusercontent.com/353044/187300852-a6532fbd-8149-47af-a508-86ebc9070124.png">

GIVEN I fill out the form on the demo page with simple alphanumeric values
WHEN I click "Sign"
**Expected** THEN my data is sent to an API 
AND my data shows up in the UI under the form with the other test data
**Actual** THEN I get the error message "Something went wrong"
AND both the GET and POST requests respond with a 500 error
AND my data doesn't show up with the other test data

**Comment by amyegan** [comment]
Thanks for letting us know @LL782! The good news is that the template still seems to be working (at least it did for me).

<img width="892" alt="image" src="https://user-images.githubusercontent.com/8247105/187520293-0eeecd89-b638-4767-b10d-53d026acd030.png">

I'll look into what's failing on the demo link and get it fixed, but please let me know if the template is giving you any problems.

**Reply by LL782** [reply]
Thanks @amyegan – I've deployed a copy of the template and it's working for me too.

I'm guess the issue is with the DB used by the demo or the secrets to connect it to Fauna

**Comment by faunaee** [comment]
Are there any error messages in your browser's console? If so, those might be helpful in diagnosing.

**Reply by LL782** [reply]
Hi @faunaee, it's a 500 Internal Server Error on both the GET and the POST to https://fauna-nextjs-guestbook-eight.vercel.app/api/entries

**Reply by faunaee** [reply]
I reviewed the demo app source a bit, and I see that the demo app prompts for the required secret but not for the domain.

Did you specify the `FAUNA_DB_DOMAIN` environment variable? If not, that could be the source of the problem. All Fauna clients must connect to the appropriate region group endpoint. See the client [connection object coverage](https://docs.fauna.com/fauna/current/drivers/connections?lang=javascript#the-client-connection-object) for details. 

If you set the `FAUNA_DB_DOMAIN` environment variable appropriately, you might find that your instance of the demo app works.

Separately, I think the 500 error is a symptom of the application not handling an error from Fauna correctly. The Fauna API can return 500 errors in some instances, but it does not do so for authentication issues. I don't know the changes that would improve the situation.

**Reply by amyegan** [reply]
Hi @faunaee! Thanks for looking into this. I think the template is still working fine. @LL782 and I were each able to set up a new deployment from the template without any trouble.

From what I could see in the in the demo app (https://fauna-nextjs-guestbook-eight.vercel.app/api/entries), a setting there may have been missing or the Fauna account used there may have been inactive. I agree with you that it would be a good idea to add some more detailed error handling!
